File: backend/app/Models/ProviderProfile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ProviderProfile extends Model
{
    use HasFactory;

    public const VERIFICATION_PENDING = 'PENDING';

    public const VERIFICATION_APPROVED = 'APPROVED';

    public const VERIFICATION_REJECTED = 'REJECTED';

    public const STATUS_ACTIVE = 'ACTIVE';

    public const STATUS_LOCKED = 'LOCKED';

    protected $fillable = [
        'user_id',
        'business_name',
        'description',
        'address',
        'identity_number',
        'experience_years',
        'average_rating',
        'total_reviews',
        'verification_status',
        'provider_status',
        'verified_at',
    ];

    protected function casts(): array
    {
        return [
            'experience_years' => 'integer',
            'average_rating' => 'decimal:2',
            'total_reviews' => 'integer',
            'verified_at' => 'datetime',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function isPending(): bool
    {
        return $this->verification_status ===
            self::VERIFICATION_PENDING;
    }

    public function isApproved(): bool
    {
        return $this->verification_status ===
            self::VERIFICATION_APPROVED;
    }

    public function isRejected(): bool
    {
        return $this->verification_status ===
            self::VERIFICATION_REJECTED;
    }

    public function isActive(): bool
    {
        return $this->isApproved()
            && $this->provider_status ===
                self::STATUS_ACTIVE;
    }

    public function isLocked(): bool
    {
        return $this->isApproved()
            && $this->provider_status ===
                self::STATUS_LOCKED;
    }
}

File: mobile/lib/models/provider_profile.dart
class ProviderProfile {
  final int id;
  final String businessName;
  final String? description;
  final String? address;
  final int experienceYears;

  final double averageRating;
  final int totalReviews;

  final String verificationStatus;
  final DateTime? verifiedAt;

  final ProviderProfileUser? user;

  const ProviderProfile({
    required this.id,
    required this.businessName,
    this.description,
    this.address,
    required this.experienceYears,
    required this.averageRating,
    required this.totalReviews,
    required this.verificationStatus,
    this.verifiedAt,
    this.user,
  });

  factory ProviderProfile.fromJson(Map<String, dynamic> json) {
    return ProviderProfile(
      id: json['id'] as int,

      businessName: json['business_name']?.toString() ?? '',

      description: json['description']?.toString(),

      address: json['address']?.toString(),

      experienceYears:
          int.tryParse(json['experience_years']?.toString() ?? '0') ?? 0,

      averageRating:
          double.tryParse(json['average_rating']?.toString() ?? '0') ?? 0,

      totalReviews: int.tryParse(json['total_reviews']?.toString() ?? '0') ?? 0,

      verificationStatus: json['verification_status']?.toString() ?? '',

      verifiedAt: json['verified_at'] == null
          ? null
          : DateTime.tryParse(json['verified_at'].toString()),

      user: json['user'] is Map
          ? ProviderProfileUser.fromJson(
              Map<String, dynamic>.from(json['user'] as Map),
            )
          : null,
    );
  }
}

class ProviderProfileUser {
  final int id;
  final String name;
  final String? email;
  final String? phone;
  final String? avatar;
  final String status;

  const ProviderProfileUser({
    required this.id,
    required this.name,
    this.email,
    this.phone,
    this.avatar,
    required this.status,
  });

  factory ProviderProfileUser.fromJson(Map<String, dynamic> json) {
    return ProviderProfileUser(
      id: json['id'] as int,
      name: json['name']?.toString() ?? '',
      email: json['email']?.toString(),
      phone: json['phone']?.toString(),
      avatar: json['avatar']?.toString(),
      status: json['status']?.toString() ?? '',
    );
  }
}
